### ISSUE TITLE
Potentially not working on xterm-256color?

### ISSUE THREAD

**Issue by lhpaoletti** [rose-pine/neovim#160]
I'm having problems with setting the `colorscheme` on Neovim to `rose-pine`. I use the Terminal app from macOS running zsh, and I can't seem to find any helpful issues here or posts on any of the goto help websites.
Maybe this could be because I don't understand Lua enough, and am doing something wrong, but what I'm doing is: make the `use`-statement in my `packer.lua`, which is in a subdirectory (which I'll call `subdir`) within the `~/.config/nvim/lua` directory. Then, from the `~/.config/nvim/init.lua` file I do `require("subdir")` at the top level. In the `subdir/init.lua` I have `require("subdir.packer")` written.
Now, when I do `:colorscheme rose-pine`, I get this horrendous look:

<img width="1062" alt="Screenshot 2023-05-23 at 01 39 52" src="https://github.com/rose-pine/neovim/assets/20807605/3e0e43b1-6a4e-4c46-91cd-c0afe2b2eea6">

As the subject of this issue says, the `$TERM` env-variable is set to `xterm-256color`. I don't know if it's relevant, but I do also have a `~/.vimrc` file.

**Comment by lhpaoletti** [comment]
I'm glad to share more information if needed. I'd be really grateful for any help, because I loved this color scheme!

**Comment by lhpaoletti** [comment]
I think not.. I just tested it with this script from [https://www.commandlinefu.com/commands/view/6533/print-all-256-colors-for-testing-term-or-for-a-quick-reference](this website) and it has the expected output for a 256-color terminal:

<img width="1162" alt="Screenshot 2023-05-26 at 17 53 10" src="https://github.com/rose-pine/neovim/assets/20807605/282769a3-ac72-4a41-b436-e4a09b14db35">

So I think that the problem might be something else here, right?

**Comment by lhpaoletti** [comment]
And I don't have an idea how to better name this issue, but maybe someone here does. I think that there should be a better, more general naming for this issue.

**Comment by mvllow** [comment]
Sorry, just to clarify is your latest screenshot in Terminal.app or another terminal? Our theme won't work in terminal.app but should in iterm2, kitty, etc. and if those are not working for you I'm happy to explore further

**Comment by syncerax** [comment]
TL;DR - Doesn't look like a bug in rose-pine. Same issue with other colorschemes. The macOS Terminal.app doesn't support 24-bit color. Switching to iTerm2 fixes the issue.

---

I ran into this as well on the macOS Terminal.app.

I first assumed that it was an issue with this colorscheme and tried out some other colorschemes. However, I saw the same problem using other colorschemes, which made me start to think that the issue was either in my configuration or in my Terminal. Messed around with my config a bit, but that didn't help either.

Also, several forums recommended switching to a different app like iTerm2, Kitty, etc. because macOS' Terminal.app doesn't have the required color support yet.

Switching to iTerm2 fixed the issue for me.

Here are some screenshots showing the rose-pine and onedark themes in iTerm2 (left) and Terminal.app (right).

|Theme|Screenshot|
|-|-|
|rose-pine|<img width="1470" alt="rose-pine" src="https://github.com/rose-pine/neovim/assets/20742562/5f6392d4-06a3-4453-912f-b37f666c020b">|
|onedark|<img width="1470" alt="onedark" src="https://github.com/rose-pine/neovim/assets/20742562/56d8bcd9-6d35-43e1-add8-bfeff65591b0">|

I think this is caused by the macOS Terminal.app not supporting **24-bit color**, and not related to `xterm-256color`. The value of the `$TERM` env variable is `xterm-256color` in both iTerm2 and Terminal.app. However, running [this script to test 24-bit color support](https://gist.github.com/lifepillar/09a44b8cf0f9397465614e622979107f) shows that iTerm2 supports 24-bit color, whereas the macOS Terminal does not. Here's a screenshot showing this:

<img width="1470" alt="24-bit-color-support" src="https://github.com/rose-pine/neovim/assets/20742562/9d35f199-84ea-4284-b98c-55dab6037430">

Here's a StackOverflow question that mentions something similar: https://stackoverflow.com/questions/58674310/nvim-in-macos-terminal-breaks-on-displaying-termguicolors

It's a little annoying, but this seems to be a macOS Terminal.app problem, not a rose-pine problem.

Hope this helps!

**Comment by mvllow** [comment]
If this is an issue outside of Terminal.app let me know and we can reopen

**Comment by mvllow** [comment]
Although it doesn't fully match our theme, it _is_ possible via [mini.colors](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-colors.md) to infer cterm values (e.g. enable support for non true color terminals):

```lua
-- Requires `rose-pine/neovim` and `echasnovski/mini.colors`
vim.cmd.colorscheme("rose-pine")
require("mini.colors").get_colorscheme():add_cterm_attributes():apply()
vim.opt.termguicolors = false
```

**Comment by ernestchu** [comment]
Actually, color will show correctly under **tmux** in Terminal.app

**Comment by mvllow** [comment]
That is not the case for my setup. Running the following in a fresh tmux instance doesn't apply my colorscheme.

```sh
tmux kill-server
tmux
nvim
```

**Comment by ernestchu** [comment]
@mvllow oh, forgot to mention. You should have this in your `~/.tmux.conf`
```
set -g default-terminal "xterm-256color"
```
